toolkit/tools/imagecustomizerapi/systemconfig.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerapi
+
+import (
+	"fmt"
+)
+
+// SystemConfig defines how each system present on the image is supposed to be configured.
+type SystemConfig struct {
+	AdditionalFiles map[string]FileConfigList `yaml:"AdditionalFiles"`
+}
+
+func (s *SystemConfig) IsValid() error {
+	var err error
+
+	for sourcePath, fileConfigList := range s.AdditionalFiles {
```

Comment:
No strong oppinions, but does it make sense to validate the source file exists for each map entry? I think it would have to happen here since the `FileConfgList` type won't have access to the `src` here.

Reply:
The `IsValid()` function doesn't take any parameters. So, it doesn't know what directory the `sourcePath` is relative to. So, it can't verify if the file exists or not.

I guess I consider `IsValid()` to be more of schema checking than full validity checking. In a future change, I have added an additional config validation to the start of the customization process. I think that would be a more appropriate place for file existence checks.

Reply:
I've added additional validity checks in `imagecustomizer.go`.